_inc/lib/admin-pages/class.jetpack-settings-page.php

```diff
@@ -55,6 +55,17 @@ function page_render() {
 			$version_notice
 		);
 
+		$rest_api_notice = str_replace(
+			'#HEADER_TEXT#',
+			esc_html( __( 'WordPress REST API is disabled', 'jetpack' ) ),
+			$rest_api_notice
+		);
+		$rest_api_notice = str_replace(
+			'#TEXT#',
+			esc_html( __( "Enable WordPress REST API to unlock Jetpack's full potential!", 'jetpack' ) ),
```

Comment:
Isn't that working for you? We are redirecting to the old settings page if REST API is disabled, aren't we?